test/studies/IMSuite/vertexColor/vertexColoring-serial.chpl

```diff
@@ -0,0 +1,321 @@
+            num_label = i;
+            node = new Node();
+            node.parent = parent[i];
+            var count = 0;
+            for j in D {
+                if(adj_graph(i,j) == 1 && node.parent != j) then count+=1;
+            }
+
+            node.neighbourD = {1..count};
+            count=1;
+
+            for j in D {
+                if(adj_graph(i,j) == 1 && node.parent != j) {
+                    node.children[count] = j;
+                    count+=1;
+                }
+            }
+            node.color = num_label;
+        }
+
+        colorLabel = log2(nodes);
+        if(1<<colorLabel < nodes) then colorLabel+=1;
+    }
+
+    /*
+        Aims to busy the threads by introducing the no-op instructions
+        equivalent to the amount of load specified.
+        :arg  weight:    Specifies the current load value for a thread.
+        :type: int(64)
+        :returns: Updated load value.
+        :rtype: int(64)
+    */
+    proc loadweight(weight: int): int {
+        var count=0;
+        for i in 0..loadValue-1 do count+=1;
+        return count+weight;
+    }
+
+    /* Runs the algorithm till the graph consists of atmost six colors. */
+    proc run() {
+        nodeSet[root].color = 0;
+        do {
+            sixColor();
+        }while(checkAgain());
+    }
+
+    /* Determines the number of different colors used in the graph.
+       :returns:    true if there is no color >= 6.
+       :rtype: bool
+    */
+    proc checkAgain(): bool {
+        var flag = || reduce again;
+        return flag;
+    }
+
+    /* Reduces the number of colors used in the graph to six. */
+    proc sixColor(){
+        forall i in D {
+            forall j in 1..nodeSet[i].children.size {
+                sendColor(nodeSet[i].children[j], nodeSet[i].color);
+            }
+
+            if(loadValue != 0) then nval[i] = loadweight(nval[i]+i);
+        }
+
+        forall i in D {
+            if(i!=root){
+                again[i]=false;
+                var xored = nodeSet[i].receivedColor ^ nodeSet[i].color;
+                for k in 0..colorLabel-1 {
+                    var pval = 1<<k;
+                    var nand :int = xored & pval;
+                    if(nand == pval) {
+                        var nxored = nodeSet[i].color & pval;
+                        if(nxored == 0) then
+                            nodeSet[i].color = 2*k + 0;
+                        else
+                            nodeSet[i].color = 2*k + 1;
+                        break;
+                    }
+                }
+
+                if(nodeSet[i].color >= 6) then again[i] = true;
+            }
+
+            if(loadValue != 0) then nval[i] = loadweight(nval[i]+i);
+        }
+    }
+
+
+   /* Reduces the number of colors from six to three. */
+   proc six2three() {
+       for k in 1..3 {
+            var x = 6-k;
+            var randStream: RandomStream = new RandomStream(3);
```

Comment:
Just noting that this likely changed out from under you since your original PR.  Sorry about that.  :(